### PR TITLE
Test the tests with a test so the tests are tested

### LINKED ISSUE #7: Test the tests
Let's test the tests with a test so the tests are tested

* Pipeline that utilizes UTF to build and test the games example

### PR DESCRIPTION
**Pull request by RothAndrew**
Closes #7

### CODE REVIEW

**Comment by RothAndrew** [comment]
The 10 second log (from `main_test.go`) isn't working for some reason, need to still look at that.

File: src/go/common.go [hunk 1]
```diff
@@ -30,3 +32,13 @@ func Teardown(t *testing.T, platform interface{}) {
 		}
 	})
 }
+
+// GetEnvVar gets an environment variable, returning an error if it isn't found.
+func GetEnvVar(varName string) (string, error) {
+	val, present := os.LookupEnv(varName)
+	if !present {
+		return "", fmt.Errorf("expected env var %v not set", varName)
```
Comment (YrrepNoj) [inline src/go/common.go:40]: We should make this a variable error so people can check `if err != nil && err != utf.ErrEnvNotFound` if not having the environment variables is ok for their situation.

File: src/go/types.go [hunk 2]
```diff
@@ -21,6 +21,6 @@ func NewEC2Platform(t *testing.T, instanceName string, region string, instanceTy
 		InstanceType: instanceType,
 		SetupScript:  setupScript,
 	}
-	platform.TestFolder = teststructure.CopyTerraformFolderToTemp(t, "..", "src/tf/public-ec2-instance")
+	platform.TestFolder = teststructure.CopyTerraformFolderToTemp(t, "../..", "src/tf/public-ec2-instance")
```
Comment (YrrepNoj) [inline src/go/types.go:24]: pretty much a nit;

but could probably use `filepaths.Join("..", "..") here

**Review by YrrepNoj — APPROVED** [review]
Could nits that you can ignore.

Looks good!

File: src/go/types.go (hunk 2 above)
Comment (RothAndrew) [reply src/go/types.go:24]: terratest docs use regular string, leaving it as is

File: src/go/common.go (hunk 1 above)
Comment (YrrepNoj) [reply src/go/common.go:40]: If you make a
```
var ErrEnvNotFound = errors.New("this is the error message")
```


you can return a `fmt.Errorf("%w: %s", varName)`


and then the caller can check for that via
`if err != nil  && errors.Is(err, utf.ErrEnvNotFound) {

}

Comment (RothAndrew) [reply src/go/common.go]: fixed

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**